File: SWE-bench/swebench/security/projects/project_g/init_db.py
"""Database initialisation — run once during Docker image build."""
import sqlite3
import bcrypt

conn = sqlite3.connect("database.db")
conn.executescript("""
CREATE TABLE IF NOT EXISTS users (
    id INTEGER PRIMARY KEY AUTOINCREMENT,
    username TEXT UNIQUE NOT NULL,
    password_hash TEXT NOT NULL,
    bio TEXT DEFAULT '',
    role TEXT DEFAULT 'user',
    api_token TEXT UNIQUE,
    webhook_url TEXT
);
CREATE TABLE IF NOT EXISTS posts (
    id INTEGER PRIMARY KEY AUTOINCREMENT,
    author_id INTEGER,
    title TEXT NOT NULL,
    body TEXT NOT NULL,
    category TEXT DEFAULT 'general'
);
""")
pw = bcrypt.hashpw(b"password123", bcrypt.gensalt()).decode()
conn.execute(
    "INSERT OR IGNORE INTO users (username, password_hash, role, api_token) "
    "VALUES ('admin', ?, 'admin', 'tok_admin_abc123')",
    (pw,),
)
conn.execute(
    "INSERT OR IGNORE INTO users (username, password_hash, api_token) "
    "VALUES ('alice', ?, 'tok_alice_xyz789')",
    (pw,),
)
conn.execute(
    "INSERT OR IGNORE INTO posts (author_id, title, body, category) "
    "VALUES (2, 'Hello World', 'First post', 'general')"
)
conn.commit()
conn.close()
print("Database initialised.")

File: SWE-bench/swebench/security/projects/project_h/init_db.js
/* init_db.js — Database initialisation, run once during Docker image build */
'use strict';

const Database = require('better-sqlite3');
const bcrypt = require('bcrypt');

const db = new Database('/app/database.db');

db.exec(`
  CREATE TABLE IF NOT EXISTS users (
    id INTEGER PRIMARY KEY AUTOINCREMENT,
    username TEXT UNIQUE NOT NULL,
    password_hash TEXT NOT NULL,
    role TEXT DEFAULT 'user',
    settings TEXT DEFAULT '{}'
  );
  CREATE TABLE IF NOT EXISTS reports (
    id INTEGER PRIMARY KEY AUTOINCREMENT,
    title TEXT NOT NULL,
    content TEXT,
    owner_id INTEGER NOT NULL
  );
  CREATE TABLE IF NOT EXISTS payments (
    id INTEGER PRIMARY KEY AUTOINCREMENT,
    user_id INTEGER NOT NULL,
    amount REAL NOT NULL,
    description TEXT
  );
`);

const hash = bcrypt.hashSync('password123', 10);
db.prepare("INSERT OR IGNORE INTO users (username, password_hash, role) VALUES ('admin', ?, 'admin')").run(hash);
db.prepare("INSERT OR IGNORE INTO users (username, password_hash) VALUES ('alice', ?)").run(hash);
db.prepare("INSERT OR IGNORE INTO reports (title, content, owner_id) VALUES ('Q1 Report', 'Revenue data...', 1)").run();

console.log('Database initialised.');
